Automattic/jetpack
# CLI: Prompt for modified projects when using `changelog add`.

#### Changes proposed in this Pull Request:

* When someone runs `jetpack changelog add`, the changelogger wizard will detect and run through the changelog wizard for each changed project.

Currently needs to be run pre-commit, otherwise it will default to regular interactive mode.

Could probably stand to be refactored and touched up, but this should be a pretty sturdy MVP.

#### Jetpack product discussion

p9dueE-2A1-p2

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* Make changes to a few different projects.
* Run `jetpack changelog add` 
* The wizard should ask if you want to add the same changelog file to all projects. Walk through the changelogger, confirm they were added correctly and staged for commit.
* Make sure all other changelog commands work fine and there are no other regressions (passthroughs should still work, single project changes should default to interactive, etc). 

#### Cases worth checking:

* If one of the projects has special changelog configs (plugins/jetpack), it should default to regular interactive mode. 
* If there's a mix of special configs and "default" projects, it should ask if you want to add the same changelog file to all default projects that it can, before defaulting to interactive for the rest.
* It should catch if existing file names exist.
* Entry shouldn't be required for patch level significance changes, but should then let you add a comment.
* If you don't want to add the same file to all projects, you walk through each project individually.
* If you don't want to do *that*, it should default to regular interactive mode.
